src/Compilers/CSharp/Portable/Symbols/Source/ThisParameterSymbol.cs

```diff
@@ -33,7 +33,7 @@ public override string Name
 
         public override TypeSymbolWithAnnotations Type
         {
-            get { return TypeSymbolWithAnnotations.Create(_containingType, isNullableIfReferenceType: false, fromDeclaration: true); }
+            get { return TypeSymbolWithAnnotations.Create(_containingType, NullableAnnotation.NotNullable); }
```

Comment:
>NotNullable [](start = 94, length = 11)

Why not `NotNullableBasedOnAnalysis`? (regardless of the type, we know `this` is not null)

Reply:
Preserving original behavior. The code is correct.

---
In reply to: [236497619](https://github.com/dotnet/roslyn/pull/31264#discussion_r236497619) [](ancestors = 236497619)

Reply:
I don't think you answered my question: Why not NotNullableBasedOnAnalysis?
Framed a different way, if I make a fix somewhere, how do I decide between NotNullable and NotNullableBasedOnAnalysis? Then how does that rationale apply here?

---
In reply to: [236778384](https://github.com/dotnet/roslyn/pull/31264#discussion_r236778384) [](ancestors = 236778384,236497619)

Reply:
> I don't think you answered my question: Why not NotNullableBasedOnAnalysis?

I think I did: "Preserving original behavior." 

> if I make a fix somewhere, 

This is not a fix, this is a refactoring that is meant to preserve original behavior.

> how do I decide between NotNullable and NotNullableBasedOnAnalysis?

I usually make the choice based on the goal of the change and desired behavior.


---
In reply to: [236782284](https://github.com/dotnet/roslyn/pull/31264#discussion_r236782284) [](ancestors = 236782284,236778384,236497619)